src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/Select.js

```diff
@@ -43,10 +44,70 @@ class Select<T> extends React.Component<Props<T>> {
 
     @observable selectedOptionRef: ?ElementRef<'li'>;
 
+    @observable buttonRefsByIndex: Map<number, ElementRef<'button'>> = new Map();
+
+    @observable searchText: string = '';
+
+    @observable focusedElementIndex: number = -1;
+
     @observable open: boolean;
 
+    @computed get buttonTextsByIndex(): Map<number, string> {
+        return Array.from(this.buttonRefsByIndex.entries())
+            .reduce((buttonTextsByIndex, [index, ref]: [number, ElementRef<'button'>]) => {
+                buttonTextsByIndex.set(index, ref.textContent);
+                return buttonTextsByIndex;
+            }, new Map());
+    }
+
+    @computed get availableButtonIndices(): number[] {
+        return Array.from(this.buttonRefsByIndex.keys());
+    }
+
+    @computed get firstSelectedIndex() {
+        let firstSelectedIndex = -1;
+
+        React.Children.forEach(this.props.children, (child: any, index: number) => {
+            if (!child || child.type !== Option) {
+                return;
+            }
+
+            if (this.props.isOptionSelected(child)) {
```

Comment:
think we need to check if `firstSelectedIndex` was already set. otherwise we will return the last one

Reply:
fixed